From c7c0848147537e969e732b398501a0b1e7fd7887 Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Mon, 21 Feb 2011 21:36:10 +0800
Subject: [PATCH 208/449] ENGR00139474 Add delay time control into loop
 decoding in vpu unit test

Add delay time control into vpu loop decoding in vpu unit test. User can
set VPU_DECODER_DELAY_MS environment variable to control the delay time.
This is helpful for checking VPU/IPU performance when bus loading is high.

Signed-off-by: Sammy He <r62914@freescale.com>
---
 test/mxc_vpu_test/dec.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/test/mxc_vpu_test/dec.c b/test/mxc_vpu_test/dec.c
index 8656c49..984ec9d 100644
--- a/test/mxc_vpu_test/dec.c
+++ b/test/mxc_vpu_test/dec.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2004-2010 Freescale Semiconductor, Inc.
+ * Copyright 2004-2011 Freescale Semiconductor, Inc.
  *
  * Copyright (c) 2006, Chips & Media.  All rights reserved.
  */
@@ -677,6 +677,7 @@ decoder_start(struct decode *dec)
 	int totalNumofErrMbs = 0;
 	int disp_clr_index = -1, actual_display_index = -1, field = V4L2_FIELD_NONE;
 	int is_waited_int = 0;
+	char *delay_ms, *endptr;
 
 	if (((dec->cmdl->dst_scheme == PATH_V4L2) || (dec->cmdl->dst_scheme == PATH_IPU))
 			&& (dec->cmdl->ipu_rot_en))
@@ -1140,6 +1141,10 @@ decoder_start(struct decode *dec)
 				return -1;
 			}
 		}
+
+	delay_ms = getenv("VPU_DECODER_DELAY_MS");
+	if (delay_ms && strtol(delay_ms, &endptr, 10))
+		usleep(strtol(delay_ms,&endptr, 10) * 1000);
 	}
 
 	if (totalNumofErrMbs) {
-- 
1.8.0

